src/AzureAuth/CommandMain.cs

```diff
@@ -262,10 +262,44 @@ private int GetToken()
             try
             {
                 ITokenFetcher tokenFetcher = this.TokenFetcher();
-                TokenResult tokenResult = (this.tokenFetcherOptions.Scopes == null
-                    ? tokenFetcher.GetAccessTokenAsync(this.CombinedAuthMode)
-                    : tokenFetcher.GetAccessTokenAsync(this.tokenFetcherOptions.Scopes, this.CombinedAuthMode))
-                    .Result;
+                TokenResult tokenResult = null;
+
+                // When running multiple AzureAuth processes with the same resource, client, and tenant IDs,
+
+                // They may prompt many times, which is annoying and unexpected.
+                // Use Mutex to ensure that only one process can access the corresponding resource at the same time.
+                string lockName = $"Global\\{this.Resource}_{this.Client}_{this.Tenant}";

```

Comment:
According to the `Mutex` documentation, using the `Global\` (or `Local\`) prefix for the name only applies to running on a Terminal Services server. From the [Remarks section of the docs](https://docs.microsoft.com/en-us/dotnet/api/system.threading.mutex?view=net-6.0#remarks):

> Note: On a server that is running Terminal Services, a named system mutex can have two levels of visibility. If its name begins with the prefix "Global\\", the mutex is visible in all terminal server sessions. If its name begins with the prefix "Local\\", the mutex is visible only in the terminal server session where it was created. In that case, a separate mutex with the same name can exist in each of the other terminal server sessions on the server. If you do not specify a prefix when you create a named mutex, it takes the prefix "Local\\". Within a terminal server session, two mutexes whose names differ only by their prefixes are separate mutexes, and both are visible to all processes in the terminal server session. That is, the prefix names "Global\\" and "Local\\" describe the scope of the mutex name relative to terminal server sessions, not relative to processes.

Not sure if you intended to restrict to only one simultaneous global auth prompt across multiple concurrent users.

Reply:
Hi @mjcheetham , thank you for your comment. Overall, yes, we do want restrict to only one auth prompt. For a same resource combination **{this.Resource}\_{this.Client}\_{this.Tenant}**, only the first successfully logged in window takes effect, the rest will use the cached authorization token.

Reply:
@mjcheetham  thanks so much for jumping in! Please do not hesitate, you are welcome here :). I agree we actually want `LOCAL\` (which if left off is the default) because of the scenario where two different logged in users are authenticating with the same triplet of GUIDs. 

It turns out the behavior of LOCAL, on Mac, also means two login shells actually don't get mutexed when they should,. So we will continue testing and might need to make this choice platform specific.

Reply:
@mjcheetham: Your uninvited commentary is actually a huge win for us. This kind of open, community-driven feedback is half the benefit of moving to GitHub for us. Thank you! ❤️

Reply:
I wonder the definition of a **session**, if a client invoke our CLI multiple times, are these CLI actually in one **session**? If so, I guess local mutex is enough for this case.

Reply:
@reillysiemens Shalini and I tested on MacOS. What we got at the time was that `Local` behaved differently on Windows and MacOS. On Windows, different PowerShells can share a local mutex, but on MacOS, multiple shells open multiple authentication prompts. If I use `Global`, the mutex works on the OS layer.

Reply:
Just to close the loop here, @mjcheetham, the final decision we landed on here was to use `Local` on all supported platforms. As you pointed out it definitely seems like the right call on Windows and has essentially the same behavior on macOS except that different login shells (and therefore different terminal windows) can prompt at the same time. Likelihood of two uncached calls from two different terminals happening at the same time is slim though, so we figured that's a reasonable risk to take on for now.